src/OddJobs/Job.hs

```diff
@@ -169,8 +169,7 @@ logCallbackErrors jid msg action = catchAny action $ \e -> log LevelError $ LogT
 
 instance HasJobRunner RunnerM where
   getPollingInterval = cfgPollingInterval . envConfig <$> ask
-  onJobFailed = cfgOnJobFailed . envConfig  <$> ask
-
+  onJobFailed = do { env <- ask; pure (cfgOnJobFailed (envConfig env)) }
```

Comment:
Just two lines below there's `cfgOnJobSuccess . envConfig <$> ask` that has not been changed. How come this line requires a change and the other one doesn't?

Reply:
It's because they have different types:

https://github.com/saurabhnanda/odd-jobs/blob/154855bd7ce1b57e48e091c27f334e3edb89db60/src/OddJobs/Types.hs#L299-L305

Beyond that, I'm not exactly sure. The `JobErrHandler` type is existentially quantified, which I think is problematic:

https://github.com/saurabhnanda/odd-jobs/blob/154855bd7ce1b57e48e091c27f334e3edb89db60/src/OddJobs/Types.hs#L110